### DISCUSSION TITLE
rendering files with ".md" extension

### DISCUSSION THREAD

**Discussion by MichelNivard** [quarto-dev/quarto-cli#8272] [Q&A] [answered]
### Description

Is there a way to quarto preview and render quarto files with a .md extension? I need the .md extension for it to retain compatibility with other tools/software. I have a workaround for render where I simply write a line of code to automatically change the filename right before I render, but I cant really find a smooth way to work qith quarto in a file with an .md extension while displaying a preview using quarto preview

**Comment by cderv** [comment]
`quarto render` on `.md` file work. Are you targetting `.qmd` file that has computation code but want to have it as `.md` in the input ? 

In that case, our engine binding may not like it indeed, as `.md` file means not computation (https://quarto.org/docs/computations/execution-options.html#engine-binding)

**Reply by MichelNivard** [reply]
Yes, sorry let me be more explicit:  I need to force .md to be bound to an engine and code in it to be evaluated, I realise this might be counter to the philosophical choices made when setting up these render conventions. So basically, can I override on .md and opt for: "engine: knitr"? Is that a reasonable feature to request or is it to much contra the design considerations made in quarto?

**Reply by MichelNivard** [reply]
The background for the request, many other tools (like the Obsidian editor/app in my case) work with .md files but are set up to ignore .qmd files or to disable certain features/niceties when editing a .qmd file. This makes it harder for me to integrate quarto fully into Obsidian.

**Reply by cderv** [reply, marked as answer]
>This makes it harder for me to integrate quarto fully into Obsidian.

First regarding Quarto and Obsidian, other related topic
* https://github.com/quarto-dev/quarto-cli/discussions/7529
* https://github.com/quarto-dev/quarto-cli/discussions/6789
* https://github.com/quarto-dev/quarto-cli/discussions/5449

There are also external resources / discussions I found at the time:

- Obsidian related discussion [forum.obsidian.md/t/quarto-support-open-source-scientific-and-technical-publishing-system/32625/36](https://forum.obsidian.md/t/quarto-support-open-source-scientific-and-technical-publishing-system/32625/36) 
- Obsidian plugin [danieltomasz/qmd-as-md-obsidian](https://github.com/danieltomasz/qmd-as-md-obsidian)

There could be ideas already and discussion on how to improve.

> can I override on .md and opt for: "engine: knitr"? 

This is not possible. The `.qmd` extension is there by design for document using Markdown syntax and quarto feature. We error is a `.md` document is using executable code. 

I believe this is against the design, and not a feature request per-se. 

> many other tools (like the Obsidian editor/app in my case) work with .md files but are set up to ignore .qmd files or to disable certain features/niceties when editing a .qmd file.

I would say this is to the other tools to account for `.qmd` syntax. They have feature for `.md` syntax, but they would also not support specific content from Quarto and would need to adapt anyway to those. Wouldn't they ?

**Reply by MichelNivard** [reply]
I had seen those threads and discussions (inspirations for what i am trying), and have been steadily building workarounds for some of the limitation within them, I have button to render obsidian files with quarto, other obsidian plugins run code chunks so I can interactively build before rendering, and I think I could get a functional live quarto preview window working, but that one is currently stuck on me needing a single file to be accepted in full by both Obsidian and quarto.  Clearly all parties involved, big organisations with design plans, have decided the file extension of flat text files matter and those are probably wise call  on both the quarto and obsidian end. My life would just become MUCH easier if either party was cool with some rule bending on file extensions ;)
